.travis.yml

```diff
@@ -45,3 +32,39 @@ matrix:
       dist: precise
       env:
         - IMAGINE_DRIVER=gmagick
+    - php: 7.2
+      env:
+        - TRAVISOPERATION=codestyle
+
+before_install:
+  - phpenv config-rm xdebug.ini || true
+
+install:
+  - |
+    set -o errexit
+    case "${TRAVISOPERATION-}" in
+        codestyle)
+            cp .travis/composer.json.codestyle ./composer.json
```

Comment:
@mlocati I would recommend of running `composer require friendsofphp/php-cs-fixer:^2.12 --dev` instead of copy composer.json or using [styleci](https://styleci.io) instead.

Reply:
I love StyleCI, but it not as configurable as php-cs-fixer. Furthermore AFAIK StyleCI doesn't support reading from `.php_cs`, so we'd need to:
- maintain the StyleCI configuration
- maintain a php-cs-fixer configuration so that developers can check/format the PHP code locally before submitting pull requests.

About using `composer require friendsofphp/php-cs-fixer:^2.12 --dev` instead of a separate `composer.json` file will increase the test time, since composer will have to install all the default packages, even if we just need `php-cs-fixer` since it's the only tool required.

Reply:
@mlocati then I would more go for the global installation over: `wget https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar -O php-cs-fixer` of the php-cs-fixer.
and as long as php-cs-fixer is not in the require-dev of the normal composer.json the contributers will not be familiar to use it.

Reply:
I agree. I updated this pull request so that developers can run php-cs-fixer simply with `composer codestyle`